src/store/edge-device.js

```diff
@@ -0,0 +1,18 @@
+import { EDGE_DEVICE_DETAILS } from './mutation-types'
+
+const state = {
+  edgeSoftwareVersion: null
+}
+
+const mutations = {
+  [EDGE_DEVICE_DETAILS] (state, edgeDetails) {
+    state.edgeSoftwareVersion = edgeDetails.version
```

Comment:
@vickywane we need some error handling for edge cases. If `edgeDetails` is undefined or doesn't have a fully populated object structure, `edgeDetails.version` will populate `state.edgeSoftwareVersion` with a value that is not user friendly. In such cases, let's make sure `state.edgeSoftwareVersion` is left undefined so that the UI components can recognize the issue and render a user friendly message.

Reply:
Alright, this makes sense.